src/OrchardCore.Modules/OrchardCore.Workflows/Http/Activities/HttpRequestTask.cs

```diff
@@ -185,14 +191,28 @@ public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecuti
             var responseCodes = ParseResponseCodes(HttpResponseCodes);
             var outcome = responseCodes.FirstOrDefault(x => x == (int)response.StatusCode);
 
-            workflowContext.LastResult = new
+            if (IsBinary) 
```

Comment:
I don't think it is necessary to have an option on the HttpRequestTask for this. The form post itself should be set with an enctype="multipart/form-data" to indicate that it is sending data.

Reply:
@Skrypt this PR is not related to form and enctype.

it is just extending HttpRequestTask to allow us to read binary content, as mentioned before, without those changes you can only fetch and read text files

Reply:
I can't find example for dotnet, but here you can read about javascript equivalent for reading binary data:

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Sending_and_Receiving_Binary_Data

Reply:
Ok, but your XMLHttpRequest here is defining the response content type. Just like a form can set the content type of its data sent. Here, you are then missing the point that this is a value that should be taken from the HttpContext itself to set the Body of the workflowContext.LastResult. Else, we will need an option for Binary, blob, document, JSON, string, empty.

Reply:
> Here, you are then missing the point that this is a value that should be taken from the HttpContext itself to set the Body of the `workflowContext.LastResult`

it is there, few lines below

![image](https://user-images.githubusercontent.com/6403130/185637296-957a04dc-032e-4045-a968-c8d5eb6b1ce8.png)

Reply:
Yeah, I saw that. But you still don't get my point.

Reply:
true, I can't get your point, but I'm sure you will help me to do so :P

again, this PR is just about adding another way of reading body passed through Http request, now you can choose if body should be read as string (text) or as byte array (binary)

> Also, you need to take into consideration that this HttpRequest can also come from an HTML form with an enctype set to "multipart/form-data".

usually, this type of requests are returning some text body like json etc.

Reply:
When you are using an `XMLHttpRequest` it is documented that it is normally possible to use the XMLHttpRequest.responseType but it seems that ASP.NET core doesn't care... it won't affect the "Accept" header of the HttpRequest.

The Accept header is meant to say that this HttpRequest will Accept a response of X content-type.

So, another way that would work is to set the "Accept" header to "application/octet-stream".

```javascript
var xhr = new XMLHttpRequest();
xhr.open('GET', '/workflows/Invoke?token=CfDJ8On1YqmVp8VLlfa7K1eDOI3SK1AVDWzbsdHVsZA6kFIS7u2v2Goh6Qjm9GzWgbEGAG_0SPBYm6tYrHR-PeyLiKQ-POdFrq3ZVOh8TIIAXfZnDuBL3wW-k78kw_jrHR-3WHGUZ6RQxE_cCbeWnetndefB8TQ4oEKIkKfvUbmdTx34EVxNeMFkyGXYx4V3NhpnVURQHYTHTH_mB4HeoD0TeW4mdamtC766Xxnzyav8-eV-', true);
xhr.responseType = 'arraybuffer'; // doesn't work with .NET
xhr.setRequestHeader("Accept", "application/octet-stream");
xhr.send(null);
```

Then, the only thing you have to is:

```c#
var acceptHeader = HttpContext.Request.Headers.TryGetValue("Accept", out var acceptValue);
```

Then return the appropriate response.content based on the value of the acceptHeader.

Reply:
And also, using this method, you can also now support multiple accepted Content Types.

Reply:
Here, also, as documented, the XMLHttpRequest.responseType will prevent that data be anything else than binary when set to "arrayBuffer". It will return null if the data sent from the server is not binary.

http://man.hubwiz.com/docset/JavaScript.docset/Contents/Resources/Documents/developer.mozilla.org/en-US/docs/Web/API/XMLHTTPRequest/responseType.html

Reply:
Or in this case here maybe we need to use "Content-Type" header if we want to only read the value of the HttpRequest. But then the "Accept" header would be for the HttpResponse.